### PR TITLE
Address vulnerabilities found by retire.js

### PR DESCRIPTION
**Pull request by marcysutton**
I did a bunch of work to get our build passing again, including updating to eslint from jshint. There were a lot of mixed spaces and tabs, and I went for tabs because our contributing document says "we prefer tabs over spaces." I used the `polyjuice` package to convert our jshint config files to eslint. There may still be some redundancies, but I already spent more time on this than I'd like.

There are still a few things to do:

- Update grunt-contrib-watch, which has an issue filed: https://github.com/gruntjs/grunt-contrib-watch/issues/554
- Check up on ignored dependencies, including `hawk`, `cryptiles`, `boom`, `sntp` and `hoek`, which are sub-dependencies of retire.js itself (which we previously added to `retireignore.json`)
- Migrate to headless Chrome from Phantomjs-prebuilt, which isn't likely to be updated

### CODE REVIEW

**Comment by marcysutton** [comment]
Flaky build! ugh. 
```
 >> Running "test-webdriver:chrome" (test-webdriver) task
    >> >> ScriptTimeoutError: asynchronous script timeout: result was not received in 60 seconds
    >> >>   (Session info: chrome=64.0.3282.140)
    >> >>   (Driver info: chromedriver=2.35.528139 (47ead77cb35ad2a9a83248b292151462a66cd881),platform=Linux 4.4.0-111-generic x86_64)
    >> Warning: Task "test-webdriver:chrome" failed. Use --force to continue.
    >> 
    >> Aborted due to warnings.
    Warning: Task "parallel:browser-test" failed. Use --force to continue.
```
Maybe someone else could try rebuilding this later, it failed for me twice in a row and is completely unrelated to these changes.

File: lib/core/utils/qsa.js [hunk 1]
```diff
@@ -73,7 +74,7 @@ function convertAttributes (atts) {
 			case '|=' :
 				regexp = new RegExp('^'+ escapeRegExp(attributeValue) +'(-|$)');
 				break;
-			case  '=' :
+			case	'=' :
```
Comment (WilcoFiers) [inline lib/core/utils/qsa.js]: That should not be a tab.

File: lib/core/utils/qsa.js [hunk 2]
```diff
@@ -88,7 +89,7 @@ function convertAttributes (atts) {
 					return attributeValue !== value;
 				};
 				break;
-			default   :
+			default	 :
```
Comment (WilcoFiers) [inline lib/core/utils/qsa.js]: Double space?

File: test/core/utils/qsa.js [hunk 3]
```diff
@@ -18,56 +18,56 @@ Vnode.prototype.getAttribute = function (att) {
...
 				children: [{
-					actualNode: new Vnode('div', 'first',[{
-						key: 'data-a11yhero',
-						value: 'faulkner'
-					}]),
+					actualNode: new Vnode('ul'),
 					shadowId: 'a',
 					children: [{
-						actualNode: new Vnode('ul'),
+						actualNode: new Vnode('li', 'breaking'),
```
Comment (WilcoFiers) [inline test/core/utils/qsa.js]: Why did this test change?

File: test/integration/rules/runner.js [hunk 4]
```diff
@@ -48,37 +48,37 @@
 							});
 
 							test[collection]
-							.forEach(function (selector) {
-								it('should find ' + JSON.stringify(selector), function () {
-									if (!nodes) {
-										assert(false, 'there are no ' + collection);
-										return;
-									}
+  							.forEach(function (selector) {
```
Comment (WilcoFiers) [inline test/integration/rules/runner.js]: You used spaces instead of a tab in this file.

**Comment by WilcoFiers** [comment]
Third time's the charm :). I'm merging, since we need this so that other PRs can pass.

**Comment by anutin544** [comment]
Ok

**Pull request merged**